### PR TITLE
Docs release 1.9.0 csi 3911 node get volume stats

### PR DESCRIPTION
**Pull request by rivkap**

### CODE REVIEW

File: docs/content/release_notes/whats_new.md [hunk 1]
```diff
@@ -8,6 +8,20 @@ IBM® block storage CSI driver 1.9.0 introduces the enhancements that are detail
...
+IBM® block storage CSI driver 1.9.0 introduces new kubelet volume metrics support. Use 
+`NodeGetVolumeStats` to retrieve volume metrics. The following metrics are currently supported:
+- kubelet_volume_stats_available_bytes
+- kubelet_volume_stats_capacity_bytes
+- kubelet_volume_stats_inodes
+- kubelet_volume_stats_inodes_free
+- kubelet_volume_stats_inodes_used
+- kubelet_volume_stats_used_bytes
+
+For more information on using kubelet volume metrics, see the [Container Storage Interface (CSI) spec documentation for `NodeGetVolumeStats`](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#nodegetvolumestats).
```
Comment (oriyarde) [inline docs/content/release_notes/whats_new.md:22]:
```suggestion
For more information about each metric, see `VolumeUsage` in the [Container Storage Interface (CSI) spec documentation for `NodeGetVolumeStats`](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#nodegetvolumestats).
```

**Review by oriyarde — COMMENTED** [review]
IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)

File: docs/content/release_notes/whats_new.md [hunk 2]
```diff
@@ -8,6 +8,20 @@ IBM® block storage CSI driver 1.9.0 introduces the enhancements that are detail
 
 Stretched volumes and stretched snapshots (FlashCopy) are now supported on SAN Volume Controller storage systems. Stretched storage topology enables disaster recovery and high availability between nodes in I/O groups at different locations. For more information, see **Product overview** > **Technical overview** > **Systems** > **Stretched systems** within the [SAN Volume Controller documentation](https://www.ibm.com/docs/en/sanvolumecontroller).
 
+## New metrics support
+
+IBM® block storage CSI driver 1.9.0 introduces new kubelet volume metric support. Use 
+`NodeGetVolumeStats` to retrieve volume metrics. The following metrics are currently supported:
```
[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/whats_new.md:14]: Is this more correct?

"Use the `NodeGetVolumeStats` Remote Procedure Call (RPC) call to retrieve the volume metrics with a metrics scraper."

[the remark this answers is not in the document]
Comment (oriyarde) [reply docs/content/release_notes/whats_new.md:14]: no, the verb "use" addresses the user, but a user cannot directly issue remote procedure calls.
possible metrics info flows:
`csi node` --gRPC--> `kubelet` --HTTP--> `user`
`csi node` --gRPC--> `kubelet` --HTTP--> `prometheus` --?--> `user`
invalid flow:
`csi node` --gRPC--> `user`
@matancarmeli7 is the feature owner, he can elaborate

[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/whats_new.md]: @matancarmeli7  / @oriyarde  Please help provide an appropriate sentence that I can work with.

Maybe more like - 
"The `NodeGetVolumeStats` Remote Procedure Call (RPC) call enables retrieval of the volume metrics with a metrics scraper."?

or maybe we don't need this sentence all together? Do we need to mention NodeGetVolumeStats?

File: docs/content/release_notes/whats_new.md (hunk 1 above)
Comment (rivkap) [reply docs/content/release_notes/whats_new.md]: updated

File: docs/content/release_notes/whats_new.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (oriyarde) [reply docs/content/release_notes/whats_new.md]: that is correct, but as your questions suggest, I agree we don't need to mention `NodeGetVolumeStats`, that is an implementation detail

**Comment by rivkap** [comment]
> IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
I based it off of stretched so it would only show the changes made for this specific feature.
i'll end up merging it into my main docs branch once i know the technical aspects are approved

File: docs/content/release_notes/whats_new.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/whats_new.md]: is the link that we provide enough information for implementation?

[the remark this answers is not in the document]
Comment (oriyarde) [reply docs/content/release_notes/whats_new.md]: yes, though I think the link I wrote [here](https://github.com/IBM/ibm-block-csi-driver/pull/460#discussion_r817495108) could be useful for how to use this feature

[the remark this answers is not in the document]
Comment (matancarmeli7) [reply docs/content/release_notes/whats_new.md]: IMO we don't need to mention `NodeGetVolumeStats`, we need to say something like this: "We are supporting exposing various metrics through `kubelet` on mounted volumes that have been created by IBM block CSI" (the english can be better, but this is the direction I think is good) and add the relevant metrics, link that Ori sent [here](https://github.com/IBM/ibm-block-csi-driver/pull/460#discussion_r817495108) and the spec link that we already provided. 
What do you think?

[the remark this answers is not in the document]
Comment (rivkap) [reply docs/content/release_notes/whats_new.md]: updated - please approve

**Comment by oriyarde** [comment]
> > IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
> > I based it off of stretched so it would only show the changes made for this specific feature.
> > i'll end up merging it into my main docs branch once i know the technical aspects are approved

it actually shows [irrelevant diff now](https://github.com/IBM/ibm-block-csi-driver/pull/460/files#diff-aa2fcc72291bf5690f95700a4df50703346a86fe5e05b50d438f4e7e94917521), and has conflicts

**Comment by rivkap** [comment]
> > > IMO the base branch for this PR should either be develop, or a general 1.9.0 docs branch (and not "stretched feature" branch)
> > > I based it off of stretched so it would only show the changes made for this specific feature.
> > > i'll end up merging it into my main docs branch once i know the technical aspects are approved
> 
> it actually shows [irrelevant diff now](https://github.com/IBM/ibm-block-csi-driver/pull/460/files#diff-aa2fcc72291bf5690f95700a4df50703346a86fe5e05b50d438f4e7e94917521), and has conflicts

I changed the branch to merge back into when I merged the stretched branch.
the **2** files that I changed are the **what's new file** and the **limitations** files. everything else is irrelevant for this specific branch. I'll resolve any conflicts before i merge into my main branch.
please provide reviews and approvals based off of these two files. thanks

**Comment by rivkap** [comment]
opening with a different PR

**Pull request closed without merging**